.github/workflows/code-checks.yml

```diff
@@ -26,17 +32,20 @@ jobs:
       uses: actions/cache@v3
       with:
         path: ~/.cache/pip
-        key: ${{ runner.os }}-pip-${{ hashFiles('requirements/pytorch/base.txt') }}
+        key: ${{ runner.os }}-pip-${{ matrix.name }}-${{ hashFiles('requirements/pytorch/base.txt') }}
         restore-keys: |
           ${{ runner.os }}-pip-
 
-    - name: Install dependencies
+    - name: Adjust dependencies
+      if: matrix.name == "pytorch"
       run: |
         pip install torch==1.12 --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
```

Comment:
Move this to the "install dependencies" step

Reply:
Then we would need to have an internal bash if/else